Sources/NestKit/GitHub/GitHubAssetRegistryClient.swift

```diff
@@ -46,6 +46,43 @@ public struct GitHubAssetRegistryClient: AssetRegistryClient {
         }
         return AssetInformation(tagName: assetResponse.tagName, assets: assets)
     }
+
+    public func fetchAssetsApplyingExcludedVersions(repositoryURL: URL, version: GitVersion, excludingVersions: [String]) async throws -> AssetInformation {
+        let assetURL = try GitHubURLBuilder.assetURL(repositoryURL, version: version, hasExcludedVersion: true)
+        var request = HTTPRequest(url: assetURL)
+        request.headerFields = [
+            .accept: "application/vnd.github+json",
+            .gitHubAPIVersion: "2022-11-28",
+        ]
+        guard let repositoryHost = repositoryURL.host() else {
+            fatalError("Unknown host")
+        }
+        if let config = registryConfigs?.config(for: repositoryURL) {
+            logger.debug("GitHub token for \(repositoryHost) is passed from \(config.environmentVariable)")
+            request.headerFields[.authorization] = "Bearer \(config.token)"
+        } else {
+            logger.debug("GitHub token for \(repositoryHost) is not provided.")
+        }
+
+        logger.debug("Request: \(repositoryURL)")
+        logger.debug("Request: \(request.headerFields)")
+        let (data, response) = try await httpClient.data(for: request)
+        logger.debug("Response: \(data.humanReadableJSONString() ?? "No data")")
+        logger.debug("Status: \(response.status)")
+
+        if response.status == .notFound {
+            throw AssetRegistryClientError.notFound
+        }
+        let assetResponses = try JSONDecoder().decode([GitHubAssetResponse].self, from: data)
```

Comment:
Based on DRY principle, could you unify this code and the code in `fetchAssets`?

Reply:
Thanks!
I created a private method called `fetchData` to share common functionality: https://github.com/mtj0928/nest/pull/33/commits/08c6a53986b07fcf107bc58da609e8287e68935a
However, it seems difficult to share the implementation with the `fetchAssets` method due to different response types.